src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -105,6 +107,8 @@ public void save(AddressBook addressBook) throws StorageOperationException {
             marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
             marshaller.marshal(toSave, fileWriter);
 
+        } catch (ReadOnlyFileSystemException rofse){
+            throw new StorageOperationException("Error writing to read-only file: " + path);
```

Comment:
In general, just showing the error message or printing the stack trace is not considered "good exception handling".
Think how you can improve the user experience in light of the exception that just occurred.